src/nvim/api/vim.c

```diff
@@ -663,6 +663,110 @@ Array nvim_get_api_info(uint64_t channel_id)
   return rv;
 }
 
+/// Call many api functions atomically
+///
+/// This has two main usages: Firstly, to perform several requests from an
+/// async context atomically, i.e. without processing requests from other rpc
+/// clients or redrawing or allowing user interaction in between. Note that api
+/// methods that could fire autocommands or do event processing still might do
+/// so. For instance invoking the :sleep command might call timer callbacks.
+/// Secondly, it can be used to reduce rpc overhead (roundtrips) when doing
+/// many requests in sequence.
+///
+/// @param calls an array of calls, where each call is described by an array
+/// with at least three elements: the request name, an array of arguments, and
+/// a boolean indicating if the return value should be sent back.
+///
+/// @return an array with three elements. The first is the number of calls that
```

Comment:
Do you mean validation of the multi_request array itself? Then an error is sent back, nothing can be returned.